File: leetcode/976. 三角形的最大周长.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        for i in range(1, len(nums) - 1):
            if nums[-i] < nums[-i - 1] + nums[-i - 2]:
                return nums[-i] + nums[-i - 1] + nums[-i - 2]
        return 0

File: leetcode/976. 三角形的最大周长.rs
impl Solution {
    pub fn largest_perimeter(nums: Vec<i32>) -> i32 {
        let mut nums = nums;
        nums.sort();
        for i in 1..=nums.len() - 2 {
            if nums[nums.len() - i] < nums[nums.len() - i - 1] + nums[nums.len() - i - 2] {
                return nums[nums.len() - i] + nums[nums.len() - i - 1] + nums[nums.len() - i - 2];
            }
        }
        0
    }
}